File: 1014-k-closest-points-to-origin/1014-k-closest-points-to-origin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        priority_queue<vector<int>, vector<vector<int>>, Comparator>pq;
        for (auto p: points) {
            pq.push({p[0] * p[0] + p[1]*p[1], p[0], p[1]});
            if (pq.size() > k) pq.pop();
        }
        vector<vector<int>> res;
        while (!pq.empty()) {
            res.push_back({pq.top()[1],pq.top()[2]});
            pq.pop();
        }
        return res;
    }

private:
struct Comparator {
    bool operator()(vector<int> &v1, vector<int> &v2) {
        return v1[0] < v2[0];
    }
};
};

File: 1014-k-closest-points-to-origin/1014-k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:

        maxH = []

        for x,y in points:
            heappush(maxH, (-1 * (x**2 + y**2), x, y))

            if len(maxH) > k:
                heappop(maxH)
        
        res = [[x,y] for _,x,y in maxH]
        return res
            
        